src/test/java/io/inholland/groep4/api/service/AccountsServiceTest.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.BOOLEAN;
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+public class AccountsServiceTest {
+
+    @Autowired
+    private UserAccountService userAccountService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    public JwtTokenProvider jwtTokenProvider;
+
+    @Test
+    public void gettingAllAccountsShouldGiveListOAccounts() {
+        List<UserAccount> accounts = userAccountService.getAllAccounts();
+
+        assertThat(accounts).isNotEmpty();
+        assertThat(accounts.get(0).getAccountBalance()).isEqualTo(500.0);
+    }
+    @Test
+    public void gettingAccountsByIdShouldGiveAccount() {
+        Long idToFind = 1l;
+        UserAccount account = userAccountService.getAccountById(idToFind);
+
+        assertThat(account).isNotNull();
+        assertThat(account.getAccountBalance()).isEqualTo(500.0);
+    }
+
+    @Test
+    public void GettingAccountWithWrongIDShouldThrowException(){
+        Long idToFind = 100l;
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            userAccountService.getAccountById(idToFind);
+        });
+
+        assertTrue(exception.getMessage().contains("Id not found"));
+    }
+
+    @Test
+    public void gettingAccountByUserShouldGiveUserAccount(){
+        String usernameToFind = "test-employee1";
+        User username = userService.findByUsername(usernameToFind);
+        List<UserAccount> userAccounts = userAccountService.getAccountsByUser(username);
+
+        assertThat(userAccounts).isNotEmpty();
+        assertThat(userAccounts.get(0).getId()).isEqualTo(6);
+    }
+
+    @Test
+    public void gettingIBAN(){
+        String IBAN = userAccountService.getIBAN();
+
+        assertThat(IBAN).isNotNull();
+        assertThat(IBAN.length()).isEqualTo(18);
+        assertThat(IBAN.contains("NL"));
+    }
+
+    @Test
+    public void AddUserAccount(){
+        User user = new User();
+        user.setUsername("son");
+        user.setPassword("goku");
+        user.setFirstName("son");
+        user.setLastName("goku");
+        user.setEmail("songoku@example.com");
+        user.setBirthdate("01/01/1975");
+
+        userService.add(user, false);
+
+        UserAccount userAccount = new UserAccount();
+        userAccount.setAccountType(UserAccount.AccountTypeEnum.CURRENT);
+        userAccount.setOwner(user);
+        userAccount.setAccountStatus(UserAccount.AccountStatusEnum.ACTIVE);
+        userAccount.setLowerLimit(100.00);
+
+        boolean ExistByIBAN = true;
+        UserAccount newUser = userAccountService.add(userAccount, ExistByIBAN);
```

Comment:
Ook de ExistByIBAN boolean die je gebruikt in de add functie snap ik niet.
Waarom heet deze zo? Je kan toch gewoon true/false gebruiken?